reasoning_gym/arithmetic/decimal_arithmetic.py

```diff
@@ -0,0 +1,114 @@
+from dataclasses import dataclass
+from random import Random
+from typing import Any, Dict, Literal, Optional
+
+from ..factory import ProceduralDataset, register_dataset
+
+
+@dataclass
+class DecimalArithmeticDatasetConfig:
+    """Configuration for decimal arithmetic dataset generation"""
+
+    num_decimal_places: int = 6
+    seed: Optional[int] = None
+    size: int = 500  # Virtual dataset size
+
+    def validate(self) -> None:
+        """Validate configuration parameters"""
+        assert self.num_decimal_places > 0, "num_decimal_places must be positive"
+
+
+def generate_arithmetic_problem(rng, num_decimal_places, operations=None):
+    """
+    Generates simple arithmetic problems with decimal numbers formatted to a specific number of decimal places.
+
+    Parameters:
+        rng
+        num_problems (int): Number of problems to generate
+        num_decimal_places (int): Number of decimal places for the numbers
+        operations (list): List of operations to use (default: ['+', '-', '*', '/'])
+
+    Returns:
+        list: List of formatted arithmetic problem strings
+    """
+    if operations is None:
+        operations = ["+", "-", "*", "/"]
+
+    max_integer_part = 10  # Maximum whole number portion before decimal
+    max_value = max_integer_part * (10**num_decimal_places)
+
+    problem = None
+
+    # Generate random numbers with exact decimal places
+    num1 = rng.randint(1, max_value) / (10**num_decimal_places)
```

Comment:
Especially interesting would be a different number of decimal places per term, like `0.1 + 0.11` .. and it would of course be cool to support more than two terms (2 could still be the default in the config).